### PR TITLE
Add: polymorphic associations section

### PR DESCRIPTION
**Pull request by thefliik**
Addresses #1432 

This pull introduces/changes:
 * adds a _Polymorphic Associations_ section to the docs

Hopefully this will save anyone else running into polymorphic association problems some time and keep them from overthinking things like I did with #1432 

Pings:
@cheerfulstoic
@subvertallchris

### CODE REVIEW

**Comment by thefliik** [comment]
Note: the code in the example will currently not work _as is_ because of #1421

File: docs/ActiveNode.rst [hunk 1]
```diff
@@ -422,6 +422,55 @@ You can even update properties of the relationships for the associations like so
...
+``has_one`` or ``has_many`` associations which target multiple ``model_class`` are called polymorphic associations.
+This is done by setting ``model_class: false`` or ``model_class: [:ModelOne, :ModelTwo, :Etc]``. In our example, the ``Person`` class has a polymorphic association ``written_things``
+
+.. code-block:: ruby
+
+    class Person
+      include Neo4j::ActiveNode
+
+      # Match all incoming relationship types
+      has_many :in, :written_things, type: false, model_class: [:Post, :Comment]
```
[the remark this answers is not in the document]
Comment (thefliik) [reply docs/ActiveNode.rst]: good point

**Comment by thefliik** [comment]
I agree. To be honest, I really don't understand `AssociationProxy` and `QueryProxy` myself yet. While I've been using them a lot for the past few months, I haven't taken the time to figure out how they work internally. Which is the primary reason why I haven't attempted to fix any of the association bugs I've encountered.

Most importantly I wonder: why is it an _association proxy_? Why not just an _association_? This is the question that keeps me up at night.

But I digress.

**Comment by cheerfulstoic** [comment]
Great, thanks!  Merged and released in 9.0.0 docs (which are now the default docs on readthedocs)

So regarding the different concepts in the gem:

Originally there was `Query` which was inspired by `ActiveRecord` and `arel` for creating queries via a Ruby DSL.  This worked pretty well, but around the same time I thought it would be cool to be able to build queries at a higher level.  It would be similar to how `ActiveRecord`'s associations lazy loaded and provided other niceties, but it could benefit from Neo4j's ability to dive deeply into relationships.  This we made `QueryProxy`.  `QueryProxy` object would wrap queries and would be returned from associations and could chain on themselves.  But after a while it became clear that we needed logic specifically around associations, particularly for handling the state of associations in-memory without needing to go to the DB all the time (also inspired by `ActiveRecord`).  This became `AssociationProxy` (the thinking was because it wrapped/replaced `QueryProxy` which retrieved from an association directly).  So the name perhaps isn't great because it's a proxy for `QueryProxy` objects, not for associations...

**Comment by cheerfulstoic** [comment]
I just realized that you said that the examples wouldn't currently work.  Was just reviewing #1421 but I probably should have read that more first.  Will continue discussion for now over there

**Pull request merged**